### PR TITLE
bullet parsing performance improvements

### PR DESCRIPTION
**Pull request by harshad1**
~~This is me experimenting with various approaches to improving performance~~

~~It may or may not result in an actual, viable, PR.~~

I think the change here is worth merging (assuming it passes all tests and I haven't missed an important case)

### CODE REVIEW

File: plugin/bullets.vim [hunk 1]
```diff
@@ -100,18 +100,50 @@ endif
...
-        \  s:match_alphabetical_list_item(a:line_text),
-        \ ],
-        \ '!empty(v:val)'
-        \ )
-
-  return s:map(l:kinds, 'extend(v:val, { "starting_at_line_num": ' . a:line_num . ' })')
+
+  let kinds = []
+
+  let bullets = s:match_bullet_list_item(a:line_text)
```
Comment (harshad1) [inline plugin/bullets.vim:106]: While much less elegant, this hierarchical approach does less work than the version it replaces. It is 2-3x faster

**Comment by harshad1** [comment]
@dkarter Any idea why the CI tests aren't running?

**Comment by dkarter** [comment]
@harshad1 sorry, not sure what happened with Travis.. looks like they changed their plans or something. I've updated my plan again to the free one and seems like I should have full credits. Maybe try again with `git commit -m 'Trigger Build' --allow-empty`

**Review by dkarter — APPROVED** [review]
Thank you! And sorry it took so long to merge (I've been very busy the past year)!

**Pull request merged**